bugsnag-plugin-android-exitinfo/src/main/java/com/bugsnag/android/EventSynthesizer.kt

```diff
@@ -0,0 +1,95 @@
+package com.bugsnag.android
+
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_CACHED
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_CANT_SAVE_STATE
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_FOREGROUND_SERVICE
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_GONE
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_PERCEPTIBLE
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_PERCEPTIBLE_PRE_26
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_SERVICE
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_TOP_SLEEPING
+import android.app.ActivityManager.RunningAppProcessInfo.IMPORTANCE_VISIBLE
+import android.app.ActivityManager.RunningAppProcessInfo.REASON_PROVIDER_IN_USE
+import android.app.ActivityManager.RunningAppProcessInfo.REASON_SERVICE_IN_USE
+import android.app.ApplicationExitInfo
+import android.os.Build
+import androidx.annotation.RequiresApi
+
+@RequiresApi(Build.VERSION_CODES.R)
+internal class EventSynthesizer(
+    private val anrEventEnhancer: (Event, ApplicationExitInfo) -> Unit,
+    private val nativeEnhancer: (Event, ApplicationExitInfo) -> Unit,
+    private val exitInfoPluginStore: ExitInfoPluginStore
+) {
+    fun createEventWithExitInfo(appExitInfo: ApplicationExitInfo): Event? {
+        val (_, exitInfoKeys) = exitInfoPluginStore.load()
+        val exitInfoKey = ExitInfoKey(appExitInfo)
+        if (appExitInfo.reason == ApplicationExitInfo.REASON_ANR &&
+            exitInfoKey !in exitInfoKeys
+        ) {
+            exitInfoPluginStore.addExitInfoKey(exitInfoKey)
```

Comment:
If we check `exitInfoKey in exitInfoKeys` earlier, we could also lift this `addExitInfoKey` out of these `if`s